flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServicesConfiguration.java

```diff
@@ -115,6 +116,8 @@ public TaskManagerServicesConfiguration(
 		this.maxJvmHeapMemory = maxJvmHeapMemory;
 		this.localRecoveryEnabled = checkNotNull(localRecoveryEnabled);
 		this.queryableStateConfig = queryableStateConfig;
+
+		Preconditions.checkArgument(0 < numberOfSlots, "The number of task slots must be greater than 0.");
```

Comment:
The commit message says the check was `moved`, but this is just one more check.
Also the null check below doesn't make sense.